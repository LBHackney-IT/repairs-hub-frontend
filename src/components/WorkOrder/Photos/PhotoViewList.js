import PhotoListWithPreview from './PhotoListWithPreview'
import PhotoGroupDescriptionForm from './PhotoGroupDescriptionForm'
import { formatDateTime } from '@/root/src/utils/time'

const PhotoViewList = ({ photos, onSubmitSetDescription = null }) => {
  // in case of error
  if (photos === null) return null

  return (
    <>
      <ul className="lbh-body-s photoViewList-ul">
        {photos.map((fileGroup) => (
          <li
            key={fileGroup.id}
            className="photoViewList-li"
            data-testid={`fileGroup-${fileGroup.id}`}
          >
            <hr className="govuk-section-break govuk-section-break--visible photoViewList-hr" />

            <>
              <div className="photoViewList-photoGroup">
                <h3>{fileGroup.groupLabel}</h3>

                <div className="govuk-!-margin-0">
                  {formatDateTime(fileGroup.timestamp)}
                </div>
              </div>
              <p className="govuk-!-margin-top-1">{fileGroup.uploadedBy}</p>

              <PhotoListWithPreview fileUrls={fileGroup.files} />
            </>

            {onSubmitSetDescription !== null ? (
              <PhotoGroupDescriptionForm
                id={fileGroup.id}
                description={fileGroup.description}
                onSubmitSetDescription={onSubmitSetDescription}
              />
            ) : (
              <>
                {fileGroup.description !== null &&
                  fileGroup.description !== '' && (
                    <p style={{ color: '#666', marginTop: '15px' }}>
                      {fileGroup.description}
                    </p>
                  )}
              </>
            )}
          </li>
        ))}
      </ul>

      <hr className="govuk-section-break govuk-section-break--visible photoViewList-hr" />
    </>
  )
}

export default PhotoViewList
