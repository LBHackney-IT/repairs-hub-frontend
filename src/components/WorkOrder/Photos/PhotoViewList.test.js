import { render } from '@testing-library/react'
import PhotoViewList from './PhotoViewList'

describe('PhotoViewList component', () => {
  it('renders nothing when photos are null', () => {
    const { asFragment } = render(<PhotoViewList photos={null} />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders list of photos', () => {
    const photos = [
      {
        id: 153,
        groupLabel: 'Closing work order',
        uploadedBy:
          'Uploaded by Dennis Reynolds (dennis.reynolds@hackney.gov.uk)',
        timestamp: '2024-08-21T13:21:37.059487Z',
        description: '',
        files: ['/mockfilepath/photo_3.jpg', '/mockfilepath/photo_2.jpg'],
      },
      {
        id: 152,
        groupLabel: 'Uploaded directly to work order',
        uploadedBy: 'Uploaded by Test Test (test.test@hackney.gov.uk)',
        timestamp: '2024-07-25T06:30:37.059487Z',
        description: 'Some description',
        files: [
          '/mockfilepath/photo_1.jpg',
          '/mockfilepath/photo_2.jpg',
          '/mockfilepath/photo_3.jpg',
        ],
      },
    ]

    const { asFragment } = render(<PhotoViewList photos={photos} />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders edit description form', () => {
    const photos = [
      {
        id: 153,
        groupLabel: 'Closing work order',
        uploadedBy:
          'Uploaded by Dennis Reynolds (dennis.reynolds@hackney.gov.uk)',
        timestamp: '2024-08-21T13:21:37.059487Z',
        description: '',
        files: ['/mockfilepath/photo_3.jpg', '/mockfilepath/photo_2.jpg'],
      },
      {
        id: 152,
        groupLabel: 'Uploaded directly to work order',
        uploadedBy: 'Uploaded by Test Test (test.test@hackney.gov.uk)',
        timestamp: '2024-07-25T06:30:37.059487Z',
        description: 'Some description',
        files: [
          '/mockfilepath/photo_1.jpg',
          '/mockfilepath/photo_2.jpg',
          '/mockfilepath/photo_3.jpg',
        ],
      },
    ]

    const { asFragment } = render(
      <PhotoViewList
        photos={photos}
        onSubmitSetDescription={() => {}}
      />
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
