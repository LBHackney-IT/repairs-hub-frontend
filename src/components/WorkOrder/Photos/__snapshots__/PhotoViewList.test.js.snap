// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhotoViewList component renders edit description form 1`] = `
<DocumentFragment>
  <ul
    class="lbh-body-s photoViewList-ul"
  >
    <li
      class="photoViewList-li"
      data-testid="fileGroup-153"
    >
      <hr
        class="govuk-section-break govuk-section-break--visible photoViewList-hr"
      />
      <div
        class="photoViewList-photoGroup"
      >
        <h3>
          Closing work order
        </h3>
        <div
          class="govuk-!-margin-0"
        >
          21 Aug 2024, 13:21
        </div>
      </div>
      <p
        class="govuk-!-margin-top-1"
      >
        Uploaded by Dennis Reynolds (dennis.reynolds@hackney.gov.uk)
      </p>
      <div
        class="photoViewList-photoGroupContainer"
      >
        <div
          class="photoViewList-photoGroupItem"
        >
          <img
            src="/mockfilepath/photo_3.jpg"
            style="width: auto; height: 150px;"
          />
        </div>
        <div
          class="photoViewList-photoGroupItem"
        >
          <img
            src="/mockfilepath/photo_2.jpg"
            style="width: auto; height: 150px;"
          />
        </div>
      </div>
      <div
        class="display-inline"
      >
        <button
          class="lbh-link updateDescriptionButton"
          type="button"
        >
          Add description
        </button>
      </div>
    </li>
    <li
      class="photoViewList-li"
      data-testid="fileGroup-152"
    >
      <hr
        class="govuk-section-break govuk-section-break--visible photoViewList-hr"
      />
      <div
        class="photoViewList-photoGroup"
      >
        <h3>
          Uploaded directly to work order
        </h3>
        <div
          class="govuk-!-margin-0"
        >
          25 Jul 2024, 06:30
        </div>
      </div>
      <p
        class="govuk-!-margin-top-1"
      >
        Uploaded by Test Test (test.test@hackney.gov.uk)
      </p>
      <div
        class="photoViewList-photoGroupContainer"
      >
        <div
          class="photoViewList-photoGroupItem"
        >
          <img
            src="/mockfilepath/photo_1.jpg"
            style="width: auto; height: 150px;"
          />
        </div>
        <div
          class="photoViewList-photoGroupItem"
        >
          <img
            src="/mockfilepath/photo_2.jpg"
            style="width: auto; height: 150px;"
          />
        </div>
        <div
          class="photoViewList-photoGroupItem"
        >
          <img
            src="/mockfilepath/photo_3.jpg"
            style="width: auto; height: 150px;"
          />
        </div>
      </div>
      <div
        class="display-inline"
      >
        <p
          style="color: rgb(102, 102, 102); margin-top: 15px;"
        >
          Some description
        </p>
        <button
          class="lbh-link updateDescriptionButton"
          type="button"
        >
          Edit description
        </button>
      </div>
    </li>
  </ul>
  <hr
    class="govuk-section-break govuk-section-break--visible photoViewList-hr"
  />
</DocumentFragment>
`;

exports[`PhotoViewList component renders list of photos 1`] = `
<DocumentFragment>
  <ul
    class="lbh-body-s photoViewList-ul"
  >
    <li
      class="photoViewList-li"
      data-testid="fileGroup-153"
    >
      <hr
        class="govuk-section-break govuk-section-break--visible photoViewList-hr"
      />
      <div
        class="photoViewList-photoGroup"
      >
        <h3>
          Closing work order
        </h3>
        <div
          class="govuk-!-margin-0"
        >
          21 Aug 2024, 13:21
        </div>
      </div>
      <p
        class="govuk-!-margin-top-1"
      >
        Uploaded by Dennis Reynolds (dennis.reynolds@hackney.gov.uk)
      </p>
      <div
        class="photoViewList-photoGroupContainer"
      >
        <div
          class="photoViewList-photoGroupItem"
        >
          <img
            src="/mockfilepath/photo_3.jpg"
            style="width: auto; height: 150px;"
          />
        </div>
        <div
          class="photoViewList-photoGroupItem"
        >
          <img
            src="/mockfilepath/photo_2.jpg"
            style="width: auto; height: 150px;"
          />
        </div>
      </div>
    </li>
    <li
      class="photoViewList-li"
      data-testid="fileGroup-152"
    >
      <hr
        class="govuk-section-break govuk-section-break--visible photoViewList-hr"
      />
      <div
        class="photoViewList-photoGroup"
      >
        <h3>
          Uploaded directly to work order
        </h3>
        <div
          class="govuk-!-margin-0"
        >
          25 Jul 2024, 06:30
        </div>
      </div>
      <p
        class="govuk-!-margin-top-1"
      >
        Uploaded by Test Test (test.test@hackney.gov.uk)
      </p>
      <div
        class="photoViewList-photoGroupContainer"
      >
        <div
          class="photoViewList-photoGroupItem"
        >
          <img
            src="/mockfilepath/photo_1.jpg"
            style="width: auto; height: 150px;"
          />
        </div>
        <div
          class="photoViewList-photoGroupItem"
        >
          <img
            src="/mockfilepath/photo_2.jpg"
            style="width: auto; height: 150px;"
          />
        </div>
        <div
          class="photoViewList-photoGroupItem"
        >
          <img
            src="/mockfilepath/photo_3.jpg"
            style="width: auto; height: 150px;"
          />
        </div>
      </div>
      <p
        style="color: rgb(102, 102, 102); margin-top: 15px;"
      >
        Some description
      </p>
    </li>
  </ul>
  <hr
    class="govuk-section-break govuk-section-break--visible photoViewList-hr"
  />
</DocumentFragment>
`;

exports[`PhotoViewList component renders nothing when photos are null 1`] = `<DocumentFragment />`;
